module/epochs/epoch_lookup_test.go

```diff
@@ -240,18 +235,25 @@ func (suite *EpochLookupSuite) TestProtocolEvents_CommittedEpoch() {
 	testEpochForViewWithFallback(suite.T(), suite.lookup, suite.state, suite.currEpoch, suite.nextEpoch)
 }
 
+// epochFixtureWithExtension creates a setup epoch with an extension.
+func (suite *EpochLookupSuite) epochFixtureWithExtension(counter, firstView, extensionFinalView uint64) (protocol.Epoch, flow.EpochExtension) {
+	setupFixture := unittest.EpochSetupFixture()
+	setupFixture.Counter = counter
+	setupFixture.FirstView = firstView
+
+	extension := flow.EpochExtension{
+		FirstView: setupFixture.FirstView,
+		FinalView: extensionFinalView,
+	}
```

Comment:
This is creating a EpochSetup and EpochExtension that wouldn't be valid in our [data model](https://github.com/onflow/flow-go/blob/eec0e768fc18ae4cbef8484b17acb15eefe5c8d9/model/flow/protocol_state.go#L59-L64). An epoch extension `X` extends an epoch `E` so that `X.FirstView = E.FinalView+1`. 

Here `E.FinalView` isn't specified (though `EpochSetupFixture` will set it to a random value), but `X.FirstView == E.FirstView` which isn't valid for any `E.FinalView` value. 

In the main business logic I suggested using the `EpochExtension` object directly to extend the epoch view range, rather than querying the protocol state. I think that change will simplify much of the mocking being done in this function (we can probably remove this function entirely).

Reply:
https://github.com/onflow/flow-go/pull/6247/commits/b2d6d94c76bfb816c665ec0c3721719bba2e42db

Reply:
https://github.com/onflow/flow-go/pull/6247/commits/efb02b6e0ff15eebd5036a2951682b41d94c72cd